addons/loyalty/models/loyalty_program.py

```diff
@@ -188,7 +188,7 @@ def _program_items_name(self):
     def _program_type_default_values(self):
         # All values to change when program_type changes
         # NOTE: any field used in `rule_ids`, `reward_ids` and `communication_plan_ids` MUST be present in the kanban view for it to work properly.
-        first_sale_product = self.env['product.product'].search([('sale_ok', '=', True)], limit=1)
+        first_sale_product = self.env['product.product'].search(['|', ('company_id', '=', False), ('company_id', 'in', self.env.companies.ids), ('sale_ok', '=', True)], limit=1)
```

Comment:
Considering that we use `self.env.company.currency_id.symbol` below, I'd say we should restrict it to `self.env.company` and not companies here.  It's only a default value anyway.